### PR TITLE
recapture the latest schema on start by recapture_schema flag

### PR DESCRIPTION
**Pull request by liulikun**
Provide a way to recapture the latest schema by `recapture_schema` config. 

**NOTE**: Only use this if there are no migrations between now and the current replicated position (from `positions` table). 

It can be used to fix schema issues.

/cc @zendesk/goanna @osheroff

### CODE REVIEW

File: src/main/java/com/zendesk/maxwell/schema/MysqlSavedSchema.java [hunk 1]
```diff
@@ -606,12 +606,6 @@ public void setSchema(Schema s) {
 		this.schema = s;
 	}
 
-	private void ensureSchemaID() {
-		if ( this.schemaID == null ) {
-			throw new RuntimeException("Can't destroy uninitialized schema!");
-		}
-	}
-
```
Comment (liulikun) [inline src/main/java/com/zendesk/maxwell/schema/MysqlSavedSchema.java]: unused code

File: src/main/java/com/zendesk/maxwell/schema/MysqlSavedSchema.java [hunk 2]
```diff
@@ -620,23 +614,6 @@ public static void delete(Connection connection, long schema_id) throws SQLExcep
...
-	public boolean schemaExists(Connection connection, long schema_id) throws SQLException {
-		if ( this.schemaID == null )
-			return false;
-		ResultSet rs = connection.createStatement().executeQuery("select id from `schemas` where id = " + schema_id);
-		return rs.next();
-	}
-
```
Comment (liulikun) [inline src/main/java/com/zendesk/maxwell/schema/MysqlSavedSchema.java]: unused code

**Comment by liulikun** [comment]
@osheroff you meant not in the `docs/config.md`? Currently it's not in `config.properties`.

File: src/main/java/com/zendesk/maxwell/MaxwellConfig.java [hunk 3]
```diff
@@ -493,6 +495,7 @@ private void setup(OptionSet options, Properties properties) {
 		this.replayMode =     fetchBooleanOption("replay", options, null, false);
 		this.masterRecovery = fetchBooleanOption("master_recovery", options, properties, false);
 		this.ignoreProducerError = fetchBooleanOption("ignore_producer_error", options, properties, true);
+		this.recaptureSchema = fetchBooleanOption("recapture_schema", options, properties, false);
```
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellConfig.java]: no, I mean here:  passing `properties` allows this into the config.properties here.

Comment (liulikun) [reply src/main/java/com/zendesk/maxwell/MaxwellConfig.java]: 👍 Good point!

File: docs/docs/config.md [hunk 4]
```diff
@@ -45,6 +45,7 @@ producer_partition_by          | [PARTITION_BY](#partition_by)       | input to
 producer_partition_columns     | STRING                              | if partitioning by 'column', a comma separated list of columns |
 producer_partition_by_fallback | [PARTITION_BY_FALLBACK](#partition_by_fallback) | required when producer_partition_by=column.  Used when the column is missing |
 ignore_producer_error          | BOOLEAN              | Maxwell will be terminated on kafka/kinesis errors when false. Otherwise, those producer errors are only logged. | true
+recapture_schema               | BOOLEAN              | recapture the latest schema. Not available in config.properties. | false
```
Comment (osheroff) [inline docs/docs/config.md]: this should be under database options, not producer options.

File: src/main/java/com/zendesk/maxwell/schema/MysqlSchemaStore.java [hunk 5]
```diff
@@ -86,6 +76,22 @@ private MysqlSavedSchema restoreOrCaptureSchema() throws SchemaStoreException {
 		}
 	}
 
+	public MysqlSavedSchema recaptureSchema() throws SQLException {
```
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/schema/MysqlSchemaStore.java]: this should just be named `captureSchema`

File: docs/docs/config.md (hunk 4 above)
Comment (liulikun) [reply docs/docs/config.md]: 👍

File: src/main/java/com/zendesk/maxwell/schema/MysqlSchemaStore.java (hunk 5 above)
Comment (liulikun) [reply src/main/java/com/zendesk/maxwell/schema/MysqlSchemaStore.java]: There is `captureSchema()` defined in the base class `AbstractSchemaStore`. Any other suggestions?

Comment (osheroff) [reply src/main/java/com/zendesk/maxwell/schema/MysqlSchemaStore.java]: `captureAndSaveSchema()` describes what it does pretty accurately.  But also you could just override `captureSchema` and have `MysqlSchemaStore#captureSchema()` do the saving as well...

Comment (liulikun) [reply src/main/java/com/zendesk/maxwell/schema/MysqlSchemaStore.java]: `captureAndSaveSchema` sounds good!

**Pull request merged**